src/dataservice/data-service-dao-file/src/main/java/com/redhat/depdraw/dataservice/dao/file/DiagramResourceDaoImpl.java

```diff
@@ -0,0 +1,81 @@
+package com.redhat.depdraw.dataservice.dao.file;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.depdraw.dataservice.dao.api.DiagramResourceDao;
+import com.redhat.depdraw.dataservice.dao.model.DiagramResource;
+
+@ApplicationScoped
+public class DiagramResourceDaoImpl implements DiagramResourceDao {
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Override
+    public DiagramResource create(DiagramResource dr) {
+        try {
+            final String s = objectMapper.writeValueAsString(dr);
+            final String pathString = FileUtil.DIAGRAM_FILES_DIR + dr.getDiagramID() + "/" + FileUtil.DIAGRAM_RESOURCES_FILES_DIR + dr.getUuid() + "/";
+            Path path = Path.of(pathString);
+
+            Files.createDirectories(path);
+
+            Path fileName = Path.of(pathString + FileUtil.DIAGRAM_RESOURCE_FILE_NAME);
+
+            // Writing into the DiagramResource file
+            Files.writeString(fileName, s, StandardOpenOption.CREATE_NEW);
+
+            fileName = Path.of(pathString + FileUtil.DIAGRAM_RESOURCE_DEFINITION_FILE_NAME);
+
+            // Writing into the DiagramResourceDefinition file
+            Files.writeString(fileName, "", StandardOpenOption.CREATE_NEW);
+        } catch (IOException e) {
+            return dr;
+        }
+
+        return dr;
+    }
```

Comment:
@tmihalac In Case there is an IOException that is caught , the file is not really created, or created corrupted , and in this case the end user/client should get 500 Internal Server error or something similar(not 200 or 201) , with some proper message that hides the low level technical details. in addition , a log error message should be printed, with error message and stack trace.

I Know great way how to do it in Quarkus, it requires you to create your custom exceptions, throw it from the caught exception block , and use Quarkus RESTEasy' ExceptionMapper Object, to map java excpetions to HTTP Response Objects:

https://howtodoinjava.com/resteasy/resteasy-exceptionmapper-example/

https://developers.redhat.com/articles/2022/03/03/rest-api-error-modeling-quarkus-20#

The pro of this technique, that is separates the logic of the error responses from the logic of the valid/normal response.

You can do it more complicated if you'll manage both the error responses and good responses  from the controller side(DataServiceResource.java)